coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java

```diff
@@ -0,0 +1,357 @@
+   * This overridden implementation removes the catalog name from the table namespace, if present
+   *
+   * @param e Input TableScan RelNode is. An example:
+   *         <pre>
+   *            LogicalTableScan(table=[[hive, default, complex]])
+   *         </pre>
+   * @return This override returns a Result whose sqlNode representation would be:
+   *         <pre>
+   *            SqlIdentifier(default, complex)
+   *         </pre>
+   *         supplemented with additional clause association details.
+   */
+  @Override
+  public Result visit(TableScan e) {
+    List<String> qualifiedName = e.getTable().getQualifiedName();
+    if (qualifiedName.size() > 2) {
+      qualifiedName = qualifiedName.subList(qualifiedName.size() - 2, qualifiedName.size()); // take last two entries
+    }
+    final SqlIdentifier identifier = new SqlIdentifier(qualifiedName, SqlParserPos.ZERO);
+    return result(identifier, ImmutableList.of(Clause.FROM), e, null);
+  }
+
+  /**
+   * Correlate relNode represents a Node with two correlated child queries,
+   * which upon conversion will be joined by non-traditional Join type.
+   * The transformation strategy for a correlate type RelNode involves independently evaluating
+   * the two sub-expressions, joining them with COMMA joinType and associating the resulting expression
+   * with the FROM clause of the SQL.
+   * The right sub-expression of a Correlate node is generally an Uncollect type RelNode. Hence, the expression
+   * obtained from evaluating this right child node will be padded with a LATERAL SqlOperator. In cases when the right
+   * sub-expression is of a different type of RelNode, ex - Project, we add the LATERAL operator on the returned Result
+   * in this step.
+   *
+   * @param e We take a RelNode of type Correlate with two child nodes as input. Example:
+   *        <pre>
+   *           LogicalCorrelate(correlation=[$cor0], joinType=[inner], requiredColumns=[{2}])
+   *                          /                                                  \
+   *    LogicalTableScan(table=[[hive, default, complex]])                   HiveUncollect
+   *                                                                               \
+   *           			          			             			          		  	LogicalProject(col=[$cor0.c])
+   *                                                                                 \
+   *           			          			            			         			 	LogicalValues(tuples=[[{ 0 }]])
+   *        </pre>
+   *
+   * @return Result of transforming the RelNode into a SqlNode with additional supplemental details.
+   *         A sample generated Result would have the following SqlNode representation:
+   *        <pre>
+   *          default.complex , LATERAL UNNEST(`complex`.`c`) AS `t0` (`ccol`)
+   *        </pre>
+   *        supplemented with additional clause association and optional alias details.
+   */
+  @Override
+  public Result visit(Correlate e) {
```

Comment:
That implementation won't be simpler. The override would need to manipulate the RightChild of the returned SqlJoin sqlNode from super only when the rightChild is of type LogicalTableFunctionScan or Uncollect. It would add some if-else logic for nested nodes. 
Plus, the current implementation keeps the logic for `visit(Correlate e)` and `visit(Join e)` in sync.